File: my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus


class HardwareStatusPublisherNode(Node):  # MODIFY NAME
    def __init__(self):
        super().__init__("Hardware_status_publisher") 
        self.hw_status_pub_ = self.create_publisher(HardwareStatus, "Hardware_status", 10)
        self.timer_ = self.create_timer(1.0, self.publish_hw_status) 
        self.get_logger().info("HW status has been published")

    def publish_hw_status (self): 
        msg = HardwareStatus()
        msg.temprature = 43.7 
        msg.are_motors_ok = True
        msg.debug_message = "Nothing special"
        self.hw_status_pub_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode()  # MODIFY NAME
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/hardware_status_publisher.cpp
    #include "rclcpp/rclcpp.hpp"
    #include "my_robot_interfaces/msg/hardware_status.hpp"


    class HardwareStatusPublisher : public rclcpp::Node // MODIFY NAME
    {
    private: 
        rclcpp::TimerBase::SharedPtr timer_ ;
        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher_;

    public:
        HardwareStatusPublisher() : Node("Hardware_Publisher") // MODIFY NAME
        {
            publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("Hardware_Status_Publisher", 10); 
            RCLCPP_INFO(this->get_logger(), "The topic has started"); 
            timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&HardwareStatusPublisher::publisher_callback, this)); 
        }
     
    private:
        void publisher_callback(){ 
            auto msg = my_robot_interfaces::msg::HardwareStatus(); 

            msg.temprature = 52.7;
            msg.are_motors_ok = true;
            msg.debug_message = "Motors are too hot";

            publisher_->publish(msg);
        }
        
    };



     
    int main(int argc, char **argv)
    {
        rclcpp::init(argc, argv);
        auto node = std::make_shared<HardwareStatusPublisher>(); // MODIFY NAME
        rclcpp::spin(node);
        rclcpp::shutdown();
        return 0;
    }